src/backend/mock/backend.test.ts

```diff
@@ -0,0 +1,90 @@
+import { WithMockBackend } from './mock';
+
+describe('(Mock Backend)', () => {
+  test('can make a request', async () => {
+    const response = JSON.stringify({ hello: 'world' });
+    const config = {
+      'latest.service->my-func': {
+        type: 'function',
+        resolve: {
+          mockedResponse: response,
+        },
+      },
+    };
+
+    const request = { body: 'test' };
+    const sd = WithMockBackend(config);
+    const actual = await sd.request('latest.service->my-func', request, {});
+    expect(actual.body).toBe(response);
+  });
+
+  test('can queue a message', async () => {
+    const response = 'abc123';
+    const config = {
+      'latest.service->my-queue': {
+        type: 'queue',
+        resolve: {
+          mockedResponse: response,
+        },
+      },
+    };
+
+    const sd = WithMockBackend(config);
+    const request = { body: 'test' };
+    const actual = await sd.queue('latest.service->my-queue', request, {});
+    expect(actual.id).toBe(response);
+  });
+
+  test('can subscribe to a queue', async () => {
+    const response = 'message';
+    const config = {
+      'latest.service->my-queue': {
+        type: 'queue',
+        resolve: {
+          mockedResponse: response,
+        },
+      },
+    };
+
+    const sd = WithMockBackend(config);
+    const request = { body: 'Test' };
+    const actual = await sd.listen('latest.service->my-queue', request);
+    actual.on('message', (message) => {
+      expect(message?.message).toBe(response);
```

Comment:
With this being asynchronous I'm not sure the test will wait until this is fired, you might have to pass `done` into the test and then execute it inside the callback here.

Reply:
Ahh ofc